src/build_meta.rs

```diff
@@ -29,15 +30,94 @@ pub enum BuildHistoryRetrieval {
     None,
 }
 
+#[derive(Debug)]
+pub struct BuildMetaWaitOptions {
+    pub sleep_on_empty: Duration,
+    pub max_empty_retries: usize,
+}
+
+impl Default for BuildMetaWaitOptions {
+    fn default() -> Self {
+        Self {
+            sleep_on_empty: Duration::from_secs(15),
```

Comment:
fine for an internal merge; but potentailly this should be more experimental backof (i assume if obs isn't very busy this can be quick)

Reply:
I believe this is *relatively* rare in the first place (though really all the sleeps in here should probably use exponential backoffs...)